scripts/calendar/generator.py

```
@@ -0,0 +1,63 @@
+        self.class_name = class_name
+        self.outfile = (
+            Path(__file__).parents[2] / "holidays" / "calendars" / f"{calendar_name}_dates.py"
+        )
+
+    def generate(self, dates: Mapping[str, Mapping[int, date | list[date]]]) -> None:
+        holiday_data = []
+        for hol_name, hol_dates in sorted(dates.items()):
+            year_dates = []
+            for year, dts in sorted(hol_dates.items()):
+                if isinstance(dts, date):
+                    dts = [dts]
```

Ruff (0.15.12)

[warning] `for` loop variable `dts` overwritten by assignment target

(PLW2901)